feel/feelalg/vectorpetsc.hpp

```diff
@@ -170,6 +171,23 @@ class VectorPetsc : public Vector<T>
     }
 
     /**
+     * @brief creates a VectorPetsc out of a VectorUblas
+     *
+     * there is no copy, PETSc will use the storage of the VectorUblas
+     */
+    template<typename Storage>
+    VectorPetsc( VectorUblas<T,Storage>& v )
+        :
+        super( v.mapPtr() ),
+        M_destroy_vec_on_exit( false )
+        {
+            VecCreateMPIWithArray( v.map().worldComm().globalComm(),
+                                   1, v.map().nLocalDofWithoutGhost(), v.map().nDof(),
+                                   std::addressof( v[0] ), &M_vec );
+            this->M_is_initialized = true;
```

Comment:
yes, it is what I mean.
There is an additional Vec M_vecLocal with the same size and ordering. It's used for the container access with local dof index (and allow copy from VectorPetsc to VectorUblas). This vector is updated from M_vec in Localize() method with MpiScatter. This strategy is old (at the begin of feel++ //) and can be rethought.